presto-main/src/test/java/com/facebook/presto/execution/resourceGroups/TestResourceGroups.java

```diff
@@ -207,30 +337,36 @@ public void testFairQueuing()
         query1a.startWaitingForPrerequisites();
         group1.run(query1a);
         assertEquals(query1a.getState(), RUNNING);
+        assertEquals(query1a.getResourceGroupQueuedOn(), Optional.of(group1.getId()));
         MockManagedQueryExecution query1b = new MockManagedQueryExecution(0);
         query1b.startWaitingForPrerequisites();
         group1.run(query1b);
         assertEquals(query1b.getState(), QUEUED);
+        assertEquals(query1b.getResourceGroupQueuedOn(), Optional.of(group1.getId()));
         MockManagedQueryExecution query1c = new MockManagedQueryExecution(0);
         query1c.startWaitingForPrerequisites();
         group1.run(query1c);
         assertEquals(query1c.getState(), QUEUED);
+        assertEquals(query1c.getResourceGroupQueuedOn(), Optional.of(group1.getId()));
         MockManagedQueryExecution query2a = new MockManagedQueryExecution(0);
         query2a.startWaitingForPrerequisites();
         group2.run(query2a);
         assertEquals(query2a.getState(), QUEUED);
+        assertEquals(query2a.getResourceGroupQueuedOn(), Optional.of(group2.getId()));
```

Comment:
I don't think the added assertions in this method are useful because it looks like we are testing the same behavior 8 times.